src/main/java-templates/org/postgresql/Driver.java

```diff
@@ -411,7 +411,7 @@ public Connection getResult(long timeout) throws SQLException {
      * @throws SQLException if the connection could not be made
      */
     private static Connection makeConnection(String url, Properties props) throws SQLException {
-        return new @JDBCCONNECTCLASS@(hostSpecs(props),
+        return new ${JDBCCONNECTCLASS}(hostSpecs(props),
```

Comment:
> And does the small adaption really outweigh the stress we have with the current pre-processing solution?

Would stress go away if we eliminate pre-processing?
In driver, we can just use
```
interface ConnectionFactory {
  Connection makeConnection(HostSpec[] hostSpecs, String user, String database, Properties info, String url)
}
```
The exact implementation can be derived out of system property or manifest property or whatever.

> would love to have an explicit name like

Explicit names are probably good, however StackOverflow is full of `org.postgresql.Driver`, so it is not that nice to drop that.

> We need a performant way of achieving the same result

@lordnelson , Do we really care of the performance of "create connection" operation? For instance, TypeInfoCache would outweigh reflection lookup in makeConnection.